### PR TITLE
Cleaned Up Loop Commands and Fixed LoopListCommand

### PR DESCRIPTION
**Pull request by ob-ab**
- Cleaned up all loop commands
- Renamed LoopListCommand to LoopCollectionCommand
- Updated LoopCollectionCommand to save loop item into an output variable
- Removed Loop.CurrentIndex as it's of no use (just like a foreach loop)
- Tested out Nested LoopCollectionCommands and it works fine
- Updated ConvertToUserVariable method in VariableMethods.cs file to parse variables of type (DataRow.ColumnName) to get DataRow values

### CODE REVIEW

File: taskt/Core/Automation/Commands/Loop Commands/LoopCollectionCommand.cs [hunk 1]
```diff
@@ -1,57 +1,63 @@
...
+        [Remarks("")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
         public string v_LoopParameter { get; set; }
 
-        public LoopListCommand()
+        [XmlAttribute]
+        [PropertyDescription("Output Collection Item Variable")]
+        [InputSpecification("Select or provide a variable from the variable list.")]
+        [SampleUsage("vUserVariable")]
+        [Remarks("If you have enabled the setting **Create Missing Variables at Runtime** then you are not required" +
```
Comment (CameronHerwig) [inline taskt/Core/Automation/Commands/Loop Commands/LoopCollectionCommand.cs:39]: Lets add a note about Datarow usage given the new functionality.

**Review by CameronHerwig — CHANGES_REQUESTED** [review]

File: taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs [hunk 2]
```diff
@@ -1,52 +1,57 @@
...
-        [Attributes.PropertyAttributes.SampleUsage("n/a")]
-        [Attributes.PropertyAttributes.Remarks("")]
+        [PropertyDescription("Additional Parameters")]
+        [InputSpecification("Select the required comparison parameters.")]
+        [SampleUsage("Param Value || {vParamValue}")]
+        [Remarks("")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
         public DataTable v_LoopActionParameterTable { get; set; }
 
         [XmlIgnore]
```
Comment (openbots-ff) [inline taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs:57]: Please make the following properties follow underscore camelcase convention

File: taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs [hunk 3]
```diff
@@ -1,52 +1,57 @@
...
         public string v_LoopActionType { get; set; }
 
         [XmlElement]
-        [Attributes.PropertyAttributes.PropertyDescription("Additional Parameters")]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
-        [Attributes.PropertyAttributes.InputSpecification("Select the required comparison parameters.")]
-        [Attributes.PropertyAttributes.SampleUsage("n/a")]
-        [Attributes.PropertyAttributes.Remarks("")]
+        [PropertyDescription("Additional Parameters")]
+        [InputSpecification("Select the required comparison parameters.")]
```
Comment (openbots-ff) [inline taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs]: Doesn't match equivalent If Command

Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs]: I've notice this is a common thing among many of your attribute changes. Please make them consistent

**Review by openbots-ff — CHANGES_REQUESTED** [review]
I think it would be a good idea to open up the equivalent If Commands next to the Loop Commands and use them as a reference for Attribute cleanup. The important thing is that they be consistent with each other since they were created similarly.

File: taskt/Core/Automation/Commands/Loop Commands/LoopCollectionCommand.cs (hunk 1 above)
Comment (ob-ab) [reply taskt/Core/Automation/Commands/Loop Commands/LoopCollectionCommand.cs:39]: done

File: taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs (hunk 2 above)
Comment (ob-ab) [reply taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs:57]: done

File: taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs (hunk 3 above)
Comment (ob-ab) [reply taskt/Core/Automation/Commands/Loop Commands/BeginLoopCommand.cs]: done

**Review by openbots-ff — DISMISSED** [review]

**Review by openbots-ff — APPROVED** [review]

**Review by CameronHerwig — APPROVED** [review]

**Pull request merged**